pkg/tfvars/openstack/openstack.go

```diff
@@ -174,14 +176,29 @@ func TFVars(masterConfigs []*v1alpha1.OpenstackProviderSpec, cloud string, exter
 		return nil, errors.Errorf("ServerGroupID is not implemented in the Installer. Please use ServerGroupName for automatic creation of the Control Plane server group.")
 	}
 
+	for _, workerConfig := range workerConfigs {
+		cfg.WorkerServerGroupNames = append(cfg.WorkerServerGroupNames, workerConfig.ServerGroupName)
+		if workerConfig.ServerGroupID != "" {
+			return nil, errors.Errorf("ServerGroupID is not implemented in the Installer. Please use ServerGroupName for automatic creation of the Compute server group.")
```

Comment:
we generally recommend the use of the resouce UUIDs in external tools by the way.
names are intended to be human readable but uuids are preferred for external orchestrateors to uses.

you are allowed to have 2 server groups in the same tenant/project with the same name which is why the uuid is preferred.

Reply:
Very true. However I could not find a way to use UUID here.

The problem is that the Installer creates the Machine and MachineSet manifests (the declarative definition of the servers to be created) _before_ it hits the OpenStack API: the UUIDs are not available at that stage because the resources do not exist yet.

Reply:
ah yes the server group create api does not allow you to choose the uuid
https://docs.openstack.org/api-ref/compute/?expanded=create-server-group-detail#create-server-group

we often allow the uuid to be specified to avoid this catch 22 situation but i guess this is not one of those cases.